src/libData/AccountData/AccountStore.cpp

```diff
@@ -179,19 +180,501 @@ void AccountStore::AddAccount(const PubKey& pubKey, const Account& account)
     AddAccount(Account::GetAddressFromPublicKey(pubKey), account);
 }
 
-void AccountStore::UpdateAccounts(const Transaction& transaction)
+bool AccountStore::UpdateAccounts(const uint64_t& blockNum,
+                                  const Transaction& transaction)
 {
-    //LOG_MARKER();
+    LOG_MARKER();
 
     const PubKey& senderPubKey = transaction.GetSenderPubKey();
     const Address fromAddr = Account::GetAddressFromPublicKey(senderPubKey);
-    const Address& toAddr = transaction.GetToAddr();
+    Address toAddr = transaction.GetToAddr();
     const uint256_t& amount = transaction.GetAmount();
 
+    bool callContract = false;
+
+    if (transaction.GetData().size() > 0 && toAddr != NullAddress)
```

Comment:
can we clear out the abnormal situations first and only handle the expected logic at the end of this function?

For example, we could write this as
```
if (transaction.GetData().empty()) {
   // report error
   return false;
}

if (toAddr != NullAddress) {
   // handle this case
}

// if (other error situations)

// Do the expected UpdateAccounts behavior 
```